core/jvm/src/main/scala/zio/ClockPlatformSpecific.scala

```diff
@@ -31,41 +30,34 @@ private[zio] trait ClockPlatformSpecific {
 
     private[this] val ConstFalse = () => false
 
-    private val maxMillis = Long.MaxValue / 1000000L
+    private final val MaxMillis = Long.MaxValue / 1000000L
 
     def asScheduledExecutorService: ScheduledExecutorService =
       service
 
     def schedule(task: Runnable, duration: Duration)(implicit unsafe: Unsafe): CancelToken =
       (duration: @unchecked) match {
         case Duration.Infinity => ConstFalse
-        case _ =>
-          val millis = duration.toMillis
+        case d if d.isZero || d.isNegative =>
+          task.run()
+          ConstFalse
+        case d =>
+          val millis = d.toMillis
           val future =
-            if (millis < maxMillis)
+            if (millis < MaxMillis)
               service.schedule(
-                new Runnable {
-                  def run: Unit =
-                    task.run()
-                },
-                duration.toNanos,
+                task,
+                d.toNanos,
```

Comment:
Why was the previous code allocating a new runnable?

Reply:
I really don't know. But it didn't seem right to me so I removed it. I can't think of a any reason why we'd want that